includes/diagnostics.php

```diff
@@ -6,15 +6,26 @@
 $dropin = $this->validate_object_cache_dropin() && ( ! defined('WP_REDIS_DISABLED') || ! WP_REDIS_DISABLED );
 
 $info[ 'Status' ] = $this->get_status();
-$info[ 'Redis Version' ] = $this->get_redis_version() ?: 'Unknown';
+
+if (defined( 'WP_REDIS_CLIENT' ) && WP_REDIS_CLIENT === 'predis' && defined( 'WP_REDIS_CLUSTER' )) {
+    $info[ 'Redis Version' ] = 'Not supported';
+} else {
+    $info[ 'Redis Version' ] = $this->get_redis_version();
+}
```

Comment:
Why is this necessary? What is the exact error you're seeing without it?

Reply:
1. The preferred condition to check whether version is available should not be determined by whether the normal value of `$this->get_redis_version()` is true or false
2. Boolean expression in the form: 
```
Boolean? True: false
```

as well as

```
$this->get_redis_version()?:"Unkwon";
```

true or false should be returned for `$this->get_redis_version()`, not other return types, and should avoid the possibility of throwing exceptions!

3. For a Boolean expression, the execution logic is that when Boolean returns a Boolean value, the code of true or false segment and its return value are executed and returned according to the Boolean value. Therefore, the true segment should not be empty!

4. Maybe my writing method is not the only one, but it can guarantee that there will be no serious problems in logic! In fact, a conforming Boolean expression should have the same effect on its equivalent code. Any conforming Boolean expression can be converted to the form of `if-else-endif` normally。

When I modified the bug, I found that redis cluster was unable to execute the server command to obtain version information. After rewriting the judgment during debugging and running normally, I decided to modify the judgment logic of the location based on the above reasons!

Reply:
It's safe cast strings as booleans: https://www.php.net/manual/en/types.comparisons.php

What's the exception you are seeing without your fix?

Reply:
It is not a matter of type conversion, but of the meaning of Boolean expressions, such as:
```
$tmp = fun() ? 1:2
```
The fun function will be called. If the return value of the function is true, the expression will return 1, otherwise it will return 2. In the original written expression, there is no true condition specified for the Boolean expression. Moreover, if the Boolean condition of the expression is function function call, once the function throws an exception, it will not execute normally!

It's been several days since the code was modified. If I want to reproduce it, I have to modify it again. It's troublesome, so I won't reproduce it. You can try it yourself! Since the later modified code can run normally and meet the specification, it is recommended to keep it!

Reply:
Why would `get_redis_version()` ever throw an exception?